app/portainer/services/authentication.js

```diff
@@ -96,12 +100,23 @@ angular.module('portainer.app').factory('Authentication', [
     }
 
     async function setUser(jwt) {
-      LocalStorage.storeJWT(jwt);
-      var tokenPayload = jwtHelper.decodeToken(jwt);
-      user.username = tokenPayload.username;
-      user.ID = tokenPayload.id;
-      user.role = tokenPayload.role;
-      await setUserTheme();
+      try {
+        LocalStorage.storeJWT(jwt);
+        var tokenPayload = jwtHelper.decodeToken(jwt);
+        user.username = tokenPayload.username;
+        user.ID = tokenPayload.id;
+        user.role = tokenPayload.role;
+        await setUserTheme();
+      } catch (error) {
+        console.log('Unable to set user', error);
+      }
+    }
+
+    function autoLoginExtension() {
+      if (window.ddExtension) {
+        console.debug('Auto-login Docker Desktop');
+        login('admin', 'Passw0rd;');
```

Comment:
maybe these should be environment variables?

Reply:
moved to const for cleaner code. Could be move to env var if requirement comes